test/model/content_test.dart

```diff
@@ -324,6 +324,58 @@ void main() {
       ]);
   });
 
+  group('parse spoilers', () {
+    testParse('with default header',
+      // ```spoiler\nhello world\n```
+      '<div class="spoiler-block"><div class="spoiler-header">\n'
+          '</div><div class="spoiler-content" aria-hidden="true">\n'
+          '<p>hello world</p>\n'
+          '</div></div>',
+      [const SpoilerNode(
+        header: [],
+        content: [ParagraphNode(links: null, nodes: [TextNode('hello world')])],
+      )],
+    );
+
+    testParse('with plain custom header',
+      // ```spoiler hello\nworld\n```
+      '<div class="spoiler-block"><div class="spoiler-header">\n'
+          '<p>hello</p>\n'
+          '</div><div class="spoiler-content" aria-hidden="true">\n'
+          '<p>world</p>\n'
+          '</div></div>',
+      [const SpoilerNode(
+        header: [ParagraphNode(links: null, nodes: [TextNode('hello')])],
+        content: [ParagraphNode(links: null, nodes: [TextNode('world')])],
+      )],
+    );
+
+    testParse('with rich header and content',
+      // ```spoiler **bold** [czo](https://chat.zulip.org/)\n*italic* [zulip](https://zulip.com/)\n```
+      '<div class="spoiler-block"><div class="spoiler-header">\n'
+          '<p><strong>bold</strong> <a href="https://chat.zulip.org/">czo</a></p>\n'
+          '</div><div class="spoiler-content" aria-hidden="true">\n'
+          '<p><em>italic</em> <a href="https://zulip.com/">zulip</a></p>\n'
+          '</div></div>',
+      [const SpoilerNode(
+        header: [ParagraphNode(
+          links: [LinkNode(url: 'https://chat.zulip.org/', nodes: [TextNode('czo')])],
+          nodes: [
```

Comment:
```suggestion
        header: [
          ParagraphNode(links: null, nodes: [
```

This convenience is in a comment on `equalsNode`, which is the engine of these parsing tests. It should probably be in `model/content_test.dart` itself so as to be more visible:
```dart
  // In [expected], for the `links` field of [ParagraphNode] or
  // any other [BlockInlineContainerNode] subclass, use `null`.
  // This field will be ignored in [expected], and instead the
  // field's value in [actual] will be checked for accuracy against
  // the [BlockInlineContainerNode.nodes] field on the same node.
  void equalsNode(ContentNode expected) {
```